nemo_automodel/components/training/step_scheduler.py

```diff
@@ -88,9 +89,18 @@ def __init__(
         if max_steps is None:
             assert self.epoch_len is not None, "epoch_len must be provided if max_steps is not provided"
             max_steps = _calculate_max_steps(self.num_epochs, self.epoch_len)
+            logger.info("max_steps not provided; will run for up to {} steps".format(max_steps))
         self.max_steps = max_steps
         assert max_steps > 0, "max_steps must be greater than 0"
 
+        if ckpt_every_steps is None:
+            if self.epoch_len is None:
+                ckpt_every_steps = self.max_steps // 2
+            else:
+                ckpt_every_steps = self.epoch_len
```

Comment:
not sure if this is needed. `is_ckpt_step` will return True both at the end of epoch and end of training as defined by max step. Imo that's sufficient